espnet2/asr/decoder/transformer_decoder.py

```diff
@@ -184,11 +184,14 @@ def score(self, ys, state, x):
         logp, state = self.forward_one_step(
             ys.unsqueeze(0), ys_mask, x.unsqueeze(0), cache=state
         )
-        return logp.squeeze(0), state
+        ## Get Attention Weights
+        attentions = self.get_attention_weights()
+
+        return logp.squeeze(0), state, attentions
 
     def batch_score(
         self, ys: torch.Tensor, states: List[Any], xs: torch.Tensor
-    ) -> Tuple[torch.Tensor, List[Any]]:
+    ) -> Tuple[torch.Tensor, List[Any], dict]:
```

Comment:
Maybe we should update ScorerInterface.batch_score like this return type to support attention weights decoding in all the models. OR if you have no plan to do that, it is OK to leave this small inconsistency.

Reply:
I think the best approach is a new NamedTuple like beam_search.Hypothesis:
```python
# in espnet.nets.scorer_interface

class BatchScore(NamedTuple):
    score: torch.Tensor
    states: List[Any] = []
    attention: dict = {}

class BatchScorerInterface:
    def batch_score(
        self, ys: torch.Tensor, states: List[Any], xs: torch.Tensor
    ) -> BatchScore:
```